contrib/babelfishpg_common/src/collation.c

```diff
@@ -119,6 +119,40 @@ coll_translate_t coll_translations[] =
 };
 #define TOTAL_COLL_TRANSLATION_COUNT (sizeof(coll_translations)/sizeof(coll_translations[0]))
 
+/*
+ * Collation translation table.  This must be sorted by the first column (BBF collation),
```

Comment:
have we explored the solution to keep collation name somewhere in catalog and later use during SQL generation

Reply:
Yes, we explored _pg_attribute.attoptions_ to store the column level collation name. But, it is complicated. 
There are few examples below which make it even more complex to handle:
1. When there are collate clause for column and check clause.
```
1> create table t1(A varchar(10) collate japanese_ci_as, check(A <> 'sflkjasdlkfjf' COLLATE sql_latin1_general_cp1_ci_as))
2> go
1> 
```
2. When there are multiple collate clause in the check constraint clause.
```
1> create table t100(a varchar, check(a <> 'sflkjasdlkfjf' COLLATE sql_latin1_general_cp1_ci_as AND a <> 'abc' COLLATE japanese_ci_as))
2> go
1> 
```

Should we go with the current approach of reverse translation for now until we find a good solution to this problem?